### PR TITLE
feat: check if ipfs is the same as resolved at pin

### PR DESCRIPTION
**Pull request by nachomazzara**
# Context

Check if IPFS is what we expect

# Issues / Tickets

[CLIENT-97](https://decentraland.atlassian.net/browse/CLIENT-97)

### CODE REVIEW

File: src/ipfs.js [hunk 1]
```diff
@@ -14,18 +14,33 @@ module.exports = class Download {
...
+        if (!isMultihash(peerId)) {
+          throw createError(400, `Invalid peerId: ${peerId}`)
+        }
         const ipns = await Ethereum.getIPNS(x, y)
         await Download.connectPeer(req.params.peerId)
         const ipfs = await Download.resolveIPNS(ipns)
+        if (ipfs !== expectedIPFS) {
+          throw createError(
+            404,
+            `The IPFS resolved doesn't match with ${expectedIPFS}. Please wait a few minutes and pin again.`
```
Comment (belohlavek) [inline src/ipfs.js]: The message could be improved, may be something like `The resolved IPFS hash doesn't match the expected IPFS hash ${expectedIPFS}. Please wait a few minutes and pin again`

File: test/ipfs.spec.js [hunk 2]
```diff
@@ -107,20 +107,37 @@ describe('IPFS', () => {
...
+
     it('should not pin files because peer has not a valid format', async () => {
       connectPeer.restore()
       const resExpected = JSON.stringify({
         error: 'Invalid peerId: peerId'
       })
-      const res = await chai.request(server).post(`/api/pin/peerId/${x}/${y}`)
+      const res = await chai
+        .request(server)
+        .post(`/api/pin/peerId/${ipfs}/${x}/${y}`)
```
Comment (belohlavek) [inline test/ipfs.spec.js]: should be ${peerId}?

File: test/test.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+I am using for testing!
```
Comment (belohlavek) [inline test/test.txt]: Should we commit this?

**Review by belohlavek — CHANGES_REQUESTED** [review]

File: test/ipfs.spec.js (hunk 2 above)
Comment (nachomazzara) [reply test/ipfs.spec.js]: Nope cause the test is `should not pin files because peer has not a valid format`

Comment (belohlavek) [reply test/ipfs.spec.js]: 👍

File: test/test.txt (hunk 3 above)
Comment (nachomazzara) [reply test/test.txt]: 🤦‍♂️

**Review by belohlavek — APPROVED** [review]

**Comment by belohlavek** [comment]
It's the IPFS key rather than the IPFS hash we are sending right?

**Pull request merged**